src/Core/Cache/DefaultCacheFactory.php

```diff
@@ -42,148 +28,97 @@ class DefaultCacheFactory implements CacheFactory
     public function __construct($args = [], LoggerInterface $logger = null)
     {
         $this->args = $args;
-        $this->logger = $logger;
+        parent::__construct($logger);
     }
 
     /**
      * @inheritdoc
      */
-    public function create($service, array $args = [])
+    public function create(string $service, array $args = []): CacheInterface
     {
         // merge args with default
         $args = array_merge($this->args, $args);
         $namespace = isset($args['namespace']) ? $args['namespace'] : '';
-        $defaultLifetime = isset($args['defaultLifetime']) ? $args['defaultLifetime'] : 0;
+        $defaultLifetime = (int) (isset($args['defaultLifetime']) ? $args['defaultLifetime'] : 0);
         $directory = isset($args['directory']) ? $args['directory'] : null;
-        $version = isset($args['version']) ? $args['version'] : null;
         $useInjector = isset($args['useInjector']) ? $args['useInjector'] : true;
 
         // In-memory caches are typically more resource constrained (number of items and storage space).
         // Give cache consumers an opt-out if they are expecting to create large caches with long lifetimes.
         $useInMemoryCache = isset($args['useInMemoryCache']) ? $args['useInMemoryCache'] : true;
+        $inMemoryCacheFactory = Environment::getEnv('SS_MEMORY_CACHEFACTORY');
 
-        // Check support
-        $apcuSupported = ($this->isAPCUSupported() && $useInMemoryCache);
-        $phpFilesSupported = $this->isPHPFilesSupported();
-
-        // If apcu isn't supported, phpfiles is the next best preference
-        if (!$apcuSupported && $phpFilesSupported) {
-            return $this->createCache(PhpFilesAdapter::class, [$namespace, $defaultLifetime, $directory], $useInjector);
+        $filesystemCache = $this->instantiateFilesystemCache($namespace, $defaultLifetime, $directory, $useInjector);
+        if (!$useInMemoryCache || !$inMemoryCacheFactory) {
+            return $this->prepareCacheForUse($filesystemCache, $useInjector);
         }
 
-        // Create filesystem cache
-        if (!$apcuSupported) {
-            return $this->createCache(
-                FilesystemAdapter::class,
-                [$namespace, $defaultLifetime, $directory],
-                $useInjector
+        // Check if SS_MEMORY_CACHEFACTORY is a factory
+        if (!is_a($inMemoryCacheFactory, InMemoryCacheFactory::class, true)) {
+            throw new LogicException(
+                'SS_MEMORY_CACHEFACTORY is not a valid InMemoryCacheFactory class name'
```

Comment:
```suggestion
                'SS_IN_MEMORY_CACHE_FACTORY environment variable is not a valid InMemoryCacheFactory class'
```

Reply:
Done, but more verbose to remove any chance of misunderstanding.